test/flutter/leak_tracking/_formatting_test.dart

```diff
@@ -0,0 +1,29 @@
+// Copyright 2023 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter_test/flutter_test.dart';
+import 'package:leak_tracker/src/leak_tracking/_formatting.dart';
+
+const _jsonEmpty = <String, dynamic>{};
+
+const libName = 'libName';
+const _json = <String, dynamic>{
+  'value': {
+    'class': {
+      'library': {'name': libName}
+    }
+  }
+};
+
+void main() {
```

Comment:
can we also test a more complicated example like in your PR description? And verify a long trace is reported correctly?

Reply:
More complicated case is tested under `test/dart`, in end to end test. And it is actually vm service bug: https://github.com/dart-lang/sdk/issues/52517

It is complexity to construct real object RetainingPath, and the constructed one will not cover real cases. So I prefer to test it in end-to-end, with really returned object. Right now the returned object is wrong. So, no test coverage yet.